internal/controllers/machine/machine_controller_test.go

```diff
@@ -2651,6 +2731,7 @@ func TestIsDeleteNodeAllowed(t *testing.T) {
 					},
 				},
 			},
+			extraMachines: []*clusterv1.Machine{cp1, cp2},
```

Comment:
why do we need this now?

Reply:
I've refactored the testing struct slightly so we can pass in zero control plane machines, otherwise without this change there are always two control plane machines provided to each test entry.